apis/krel/v1alpha1/types.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// PlatformType is an OS / architecture tuple.
+type PlatformType string
+
+// Platform types.
+const (
+	PlatformTypeLinuxAMD64   PlatformType = "linux_amd64"
+	PlatformTypeDarwinAMD64  PlatformType = "darwin_amd64"
+	PlatformTypeWindoesAMD64 PlatformType = "windows_amd64"
+	PlatformTypeLinux386     PlatformType = "linux_386"
+	PlatformTypeWindows386   PlatformType = "windows_386"
+	PlatformTypeLinuxARM     PlatformType = "linux_arm"
+	PlatformTypeLinuxARM64   PlatformType = "linux_arm64"
+	PlatformTypeLinuxPPC64LE PlatformType = "linux_ppc64le"
+	PlatformTypeLinuxS390X   PlatformType = "linux_s390x"
+)
```

Comment:
I think we should do a quick brainstorm to try to find all possible uses of the arch/platform/labels/etc structs and catalogs to avoid overlapping data structures. For example, I have a similar set of labels here which I could rewrite another if we standardize:
https://github.com/kubernetes/release/blob/5b258d8961ca446353cca43e6f5f6653b9a3eefa/pkg/binary/binary.go#L25-L40

Let's consolidate them while the codebase is still small.

Reply:
@puerco SGTM :) I was thinking about developing a `krel` specific pattern, but that is probably an over-optimization and we should likely just standardize on Go platform tuples. What do you think?